### PR TITLE
[kitcli] Standardized naming and reorganized reconcilers

### PR DESCRIPTION
**Pull request by ellistarn**
Issue #, if available:

Description of changes:
- Separates controllers into cluster/infrastructure
- Standardized naming

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: substrate/pkg/controller/substrate/cluster/instanceprofile.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		if err.(awserr.Error).Code() != iam.ErrCodeNoSuchEntityException {
+			return reconcile.Result{}, fmt.Errorf("deleting instance profile %w,", err)
+		}
+	} else {
+		logging.FromContext(ctx).Infof("Deleted instance profile %s", aws.StringValue(discovery.Name(substrate)))
+	}
+	// Role
+	if _, err := i.IAM.DeleteRoleWithContext(ctx, &iam.DeleteRoleInput{RoleName: discovery.Name(substrate)}); err != nil {
+		if err.(awserr.Error).Code() != iam.ErrCodeNoSuchEntityException {
+			return reconcile.Result{}, fmt.Errorf("deleting, %w", err)
```
Comment (prateekgogia) [inline substrate/pkg/controller/substrate/cluster/instanceprofile.go:146]: nit: "deleting role"

File: substrate/pkg/controller/substrate/infrastructure/routetable.go [hunk 2]
```diff
@@ -87,10 +88,10 @@ func (r *routeTable) Delete(ctx context.Context, substrate *v1alpha1.Substrate)
 	return reconcile.Result{}, nil
 }
 
-func privateTableName(identifier string) string {
-	return fmt.Sprintf("%s-%s", identifier, "private")
+func privateTableName(substrate *v1alpha1.Substrate) string {
+	return fmt.Sprintf("%s-%s", substrate.Name, "private")
 }
```
Comment (prateekgogia) [inline substrate/pkg/controller/substrate/infrastructure/routetable.go:93]: Should we be using the same pattern here as in subnets?
```suggestion
func routeTableName(substrate *v1alpha1.Substrate, public bool) *string {
	if public {
		return discovery.Name(substrate, "public")
	}
	return discovery.Name(substrate, "private")
}
```

File: substrate/pkg/controller/substrate/cluster/instanceprofile.go (hunk 1 above)
Comment (ellistarn) [inline substrate/pkg/controller/substrate/cluster/instanceprofile.go:146]:
```suggestion
			return reconcile.Result{}, fmt.Errorf("deleting role, %w", err)
```

**Review by prateekgogia — APPROVED** [review]
Minor cleanup needed in routetable file, we can address it in next PR.

**Pull request merged**